### PR TITLE
fs/accounting: sort transfers by start time

### PR DESCRIPTION
**Pull request by Max-Sum**
#### What is the purpose of this change?

Sort transfers in accounting by start time.

#### Was the change discussed in an issue or in the forum before?

https://forum.rclone.org/t/order-of-statistics/17128/5

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: fs/accounting/stats.go [hunk 1]
```diff
@@ -558,8 +551,9 @@ func (s *StatsInfo) RetryAfter() time.Time {
 
 // NewCheckingTransfer adds a checking transfer to the stats, from the object.
 func (s *StatsInfo) NewCheckingTransfer(obj fs.Object) *Transfer {
-	s.checking.add(obj.Remote())
-	return newCheckingTransfer(s, obj)
+	tr := newCheckingTransfer(s, obj)
```
Comment (ncw) [inline fs/accounting/stats.go:554]: What do all these changes do? They don't appear to be sorting transfers by start time? Do they need to go into another commit?

Comment (Max-Sum) [reply fs/accounting/stats.go:554]: No. This have to do with stringSet -> transferMap changes. If you need more commits, I could seperate them.

Comment (ncw) [reply fs/accounting/stats.go:554]: Can you separate them, please? I'd like to have one commit with just the sorting changes in just in case I need to revert it.

Thank you :-)

Comment (Max-Sum) [reply fs/accounting/stats.go:554]: Or should I just put all stringSet->transferMap in the first commit?

**Comment by ncw** [comment]
Sorry @Max-Sum I lost track of this PR. Do you think it is ready to merge now?

**Comment by Max-Sum** [comment]
I think it's ok.

**Comment by Max-Sum** [comment]
> 
> 
> Sorry @Max-Sum I lost track of this PR. Do you think it is ready to merge now?

I think it's ok.

**Comment by ncw** [comment]
OK I'll merge it now - thank you :-)
